pkg/kubelet/eviction/eviction_manager.go

```diff
@@ -406,7 +409,14 @@ func (m *managerImpl) synchronize(diskInfoProvider DiskInfoProvider, podFunc Act
 	for i := range activePods {
 		pod := activePods[i]
 		gracePeriodOverride := int64(0)
+		var semaphore chan bool
 		if !isHardEvictionThreshold(thresholdToReclaim) {
+			semaphore = m.softEvictionSemaphore
+			if len(m.softEvictionSemaphore) != 0 {
+				klog.InfoS("Eviction manager: soft eviction already in progress, will not soft evict another pod")
+				return nil, nil
+			}
+			m.softEvictionSemaphore <- true
```

Comment:
If a pod is currently being evicted, i.e. the semaphore is not empty, this will be blocking, which I think is not the desired behavior. For example, it can lead to all of the controller's threads to get stuck here hence when a hard eviction threshold will be met the controller wouldn't be able to address it. Do you agree?

Is there a reason to use a semaphore here?
Perhaps we can use a simple lock instead, and if we see that a soft eviction threshold is taking place and hard thresholds aren't met - just return.

Does that make sense? Am I missing something?

Reply:
Oh yeah, a simple lock does sound better, let me do that